Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocViewController.swift

```diff
@@ -0,0 +1,374 @@
+            smallEventCell.configure(viewModel: smallEvent, theme: theme)
+            cell = smallEventCell
+        }
+        
+        cell.layoutMargins = layout.itemLayoutMargins
+        estimate.height = cell.sizeThatFits(CGSize(width: columnWidth, height: UIView.noIntrinsicMetric), apply: false).height
+        estimate.precalculated = true
+        
+        return estimate
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        
+        guard let articleAsLivingDocViewModel = delegate?.articleAsLivingDocViewModel else {
+            return
+        }
+
+        (cell as? ArticleAsLivingDocLargeEventCollectionViewCell)?.delegate = self
+        (cell as? ArticleAsLivingDocLargeEventCollectionViewCell)?.articleDelegate = self
+        (cell as? ArticleAsLivingDocSmallEventCollectionViewCell)?.delegate = self
+        
+        let numSections = dataSource.numberOfSections(in: collectionView)
+        let numEvents = dataSource.collectionView(collectionView, numberOfItemsInSection: indexPath.section)
+        
+        if indexPath.section == numSections - 1 &&
+            indexPath.item == numEvents - 1 {
+            guard let nextRvStartId = articleAsLivingDocViewModel.nextRvStartId,
+                  nextRvStartId != 0 else {
+                return
+            }
+            
+            delegate?.fetchNextPage(nextRvStartId: nextRvStartId)
+        }
+    }
+    
+    @objc func collectionView(_ collectionView: UICollectionView, shouldHighlightItemAt indexPath: IndexPath) -> Bool {
+        return false
+    }
+    
+    @objc func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        return false
+    }
+
+    override func collectionViewFooterButtonWasPressed(_ collectionViewFooter: CollectionViewFooter) {
+        tappedViewFullHistoryButton()
+    }
+}
+
+// MARK:- ArticleAsLivingDocHorizontallyScrollingCellDelegate
+@available(iOS 13.0, *)
+extension ArticleAsLivingDocViewController: ArticleAsLivingDocHorizontallyScrollingCellDelegate {
+    func tappedLink(_ url: URL, cell: ArticleAsLivingDocHorizontallyScrollingCell?, sourceView: UIView, sourceRect: CGRect?) {
+        if url.absoluteString.removingPercentEncoding?.contains("/User:") == true {
+            // User talk page, should open it in a modal
+            let singlePageWebVC = SinglePageWebViewController(url: url, theme: theme, doesUseSimpleNavigationBar: true)
+            let navController = WMFThemeableNavigationController(rootViewController: singlePageWebVC, theme: theme)
+            navController.modalPresentationStyle = .pageSheet
+            self.present(navController, animated: true, completion: nil)
+        } else {
+            if let linkURL = delegate?.articleURL.resolvingRelativeWikiHref(url.absoluteString) {
+                switch Configuration.current.router.destination(for: linkURL) {
```

Comment:
@mcleinman If the goal is to push to their user talk page, we also might be able to format the `User:` link in a particular way that the router expects (within `userInfoForTraitCollection` in `ArticleAsLivingDocViewModels`, so that you don't have to explicitly inspect the url `absoluteString`. The router should already be set up to detect talk page urls and push the native talk page view.

Reply:
We're (currently, could change after today's discussion) handling the user page by presenting it modally on top of AALD. Other types get a push from the main article page. So we need to determine if it's a user page to determine if we need to dismiss the AALD modal before handling the link.